--- a/hack/lib/golang.sh
+++ b/hack/lib/golang.sh
@@ -268,11 +268,27 @@ kube::golang::set_platform_envs() {
     case "${platform}" in
       "linux/arm")
         export CGO_ENABLED=1
-        export CC=arm-linux-gnueabihf-gcc
+        export CC=arm-openwrt-linux-gcc
         ;;
       "linux/arm64")
         export CGO_ENABLED=1
-        export CC=aarch64-linux-gnu-gcc
+        export CC=arm-openwrt-linux-gcc
+        ;;
+      "linux/mipsle")
+        export CGO_ENABLED=1
+        export CC=mipsel-openwrt-linux-gcc
+        ;;
+      "linux/mips")
+        export CGO_ENABLED=1
+        export CC=mips-openwrt-linux-gcc
+        ;;
+      "linux/mips64")
+        export CGO_ENABLED=1
+        export CC=mips64-openwrt-linux-gcc
+        ;;
+      "linux/mips64le")
+        export CGO_ENABLED=1
+        export CC=mipsel64-openwrt-linux-gcc
         ;;
       "linux/ppc64le")
         export CGO_ENABLED=1
