--- a/src/make.bash
+++ b/src/make.bash
@@ -131,7 +131,22 @@ if [ "$GOROOT_BOOTSTRAP" == "$GOROOT" ];
 	exit 1
 fi
 rm -f cmd/dist/dist
-GOROOT="$GOROOT_BOOTSTRAP" GOOS="" GOARCH="" "$GOROOT_BOOTSTRAP/bin/go" build -o cmd/dist/dist ./cmd/dist
+
+echo ${GO_VERSION}
+if [ ! "$GO_VERSION" = "1.8.5" ] && [ ! "$GO_VERSION" = "1.8.7" ]; then
+	if [ -z "$GOMIPS" ] || [ ! "$GOMIPS" = "softfloat" ]; then
+		GOROOT="$GOROOT_BOOTSTRAP" GOMIPS="hardfloat" GOOS="" GOARCH="" "$GOROOT_BOOTSTRAP/bin/go" build -o cmd/dist/dist ./cmd/dist
+	else
+		GOROOT="$GOROOT_BOOTSTRAP" GOMIPS="softfloat" GOOS="" GOARCH="" "$GOROOT_BOOTSTRAP/bin/go" build -o cmd/dist/dist ./cmd/dist
+	fi
+else
+	if [ ! -z "$GOMIPS" ] && [ "$GOMIPS" = "softfloat" ]; then
+		GOROOT="$GOROOT_BOOTSTRAP" GOMIPS="softfloat" GOOS="" GOARCH="" "$GOROOT_BOOTSTRAP/bin/go" build -o cmd/dist/dist ./cmd/dist
+	else
+		GOROOT="$GOROOT_BOOTSTRAP" GOMIPS="$GOMIPS" GOOS="" GOARCH="" "$GOROOT_BOOTSTRAP/bin/go" build -o cmd/dist/dist ./cmd/dist
+	fi
+fi
+# GOROOT="$GOROOT_BOOTSTRAP" GOOS="" GOARCH="" "$GOROOT_BOOTSTRAP/bin/go" build -o cmd/dist/dist ./cmd/dist
 
 # -e doesn't propagate out of eval, so check success by hand.
 eval $(./cmd/dist/dist env -p || echo FAIL=true)
@@ -156,7 +171,21 @@ if [ "$1" = "--no-clean" ]; then
 	buildall=""
 	shift
 fi
-./cmd/dist/dist bootstrap $buildall $GO_DISTFLAGS -v # builds go_bootstrap
+
+if [ ! "$GO_VERSION" = "1.8.5" ] && [ ! "$GO_VERSION" = "1.8.7" ]; then
+	if [ -z "$GOMIPS" ] || [ ! "$GOMIPS" = "softfloat" ]; then
+		GOMIPS="hardfloat" ./cmd/dist/dist bootstrap $buildall $GO_DISTFLAGS -v # builds go_bootstrap
+	else
+		GOMIPS="softfloat" ./cmd/dist/dist bootstrap $buildall $GO_DISTFLAGS -v # builds go_bootstrap
+	fi
+else
+	if [ ! -z "$GOMIPS" ] && [ "$GOMIPS" = "softfloat" ]; then
+		GOMIPS="softfloat" ./cmd/dist/dist bootstrap $buildall $GO_DISTFLAGS -v # builds go_bootstrap
+	else
+		GOMIPS="$GOMIPS" ./cmd/dist/dist bootstrap $buildall $GO_DISTFLAGS -v # builds go_bootstrap
+	fi
+fi
+# ./cmd/dist/dist bootstrap $buildall $GO_DISTFLAGS -v # builds go_bootstrap
 
 # Delay move of dist tool to now, because bootstrap may clear tool directory.
 mv cmd/dist/dist "$GOTOOLDIR"/dist
