--- a/Makefile
+++ b/Makefile
@@ -13,7 +13,7 @@ $( shell mkdir -p bin )
 user=$(shell id -u -n)
 group=$(shell id -g -n)
 
-export GOBIN=$(PWD)/bin
+# export GOBIN=$(PWD)/bin
 
 LD_FLAGS="-w -X $(REPO_PATH)/version.Version=$(VERSION)"
 
@@ -30,7 +30,7 @@ bin/grpc-client: check-go-version
 
 .PHONY: release-binary
 release-binary:
-	@go build -o /go/bin/dex -v -ldflags $(LD_FLAGS) $(REPO_PATH)/cmd/dex
+	@go build -o bin/dex -v -ldflags $(LD_FLAGS) $(REPO_PATH)/cmd/dex
 
 .PHONY: revendor
 revendor: bin/license-bill-of-materials
