--- a/Makefile.linux
+++ b/Makefile.linux
@@ -18,7 +18,7 @@ COMMANDS += containerd-shim
 
 # check GOOS for cross compile builds
 ifeq ($(GOOS),linux)
-	GO_GCFLAGS += -buildmode=pie
+	GO_GCFLAGS += -buildmode=default
 endif
 
 # amd64 supports go test -race
--- a/Makefile
+++ b/Makefile
@@ -17,7 +17,7 @@
 ROOTDIR=$(dir $(abspath $(lastword $(MAKEFILE_LIST))))
 
 # Base path used to install.
-DESTDIR=/usr/local
+DESTDIR=$(GOPATH)
 
 # Used to populate variables in version package.
 VERSION=$(shell git describe --match 'v[0-9]*' --dirty='.m' --always)
@@ -207,8 +207,8 @@ clean: ## clean up binaries
 
 install: ## install binaries
 	@echo "$(WHALE) $@ $(BINARIES)"
-	@mkdir -p $(DESTDIR)/bin
-	@install $(BINARIES) $(DESTDIR)/bin
+	@mkdir -p $(DESTDIR)/$(GO_PKG_BIN_PATH)
+	@install $(BINARIES) $(DESTDIR)/$(GO_PKG_BIN_PATH)
 
 uninstall:
 	@echo "$(WHALE) $@"
