--- a/cmd/ctr/commands/signal_map_linux.go
+++ b/cmd/ctr/commands/signal_map_linux.go
@@ -42,7 +42,6 @@ var signalMap = map[string]syscall.Signa
 	"PWR":    unix.SIGPWR,
 	"QUIT":   unix.SIGQUIT,
 	"SEGV":   unix.SIGSEGV,
-	"STKFLT": unix.SIGSTKFLT,
 	"STOP":   unix.SIGSTOP,
 	"SYS":    unix.SIGSYS,
 	"TERM":   unix.SIGTERM,
--- a/vendor/github.com/docker/docker/pkg/signal/signal_linux.go
+++ b/vendor/github.com/docker/docker/pkg/signal/signal_linux.go
@@ -32,7 +32,6 @@ var SignalMap = map[string]syscall.Signa
 	"PWR":      unix.SIGPWR,
 	"QUIT":     unix.SIGQUIT,
 	"SEGV":     unix.SIGSEGV,
-	"STKFLT":   unix.SIGSTKFLT,
 	"STOP":     unix.SIGSTOP,
 	"SYS":      unix.SIGSYS,
 	"TERM":     unix.SIGTERM,
