--- a/vendor/github.com/docker/docker/pkg/system/stat_linux.go
+++ b/vendor/github.com/docker/docker/pkg/system/stat_linux.go
@@ -8,7 +8,7 @@ func fromStatT(s *syscall.Stat_t) (*Stat
 		mode: s.Mode,
 		uid:  s.Uid,
 		gid:  s.Gid,
-		rdev: s.Rdev,
+		rdev: uint64(s.Rdev),
 		mtim: s.Mtim}, nil
 }
 
--- a/vendor/github.com/opencontainers/runc/libcontainer/devices/devices.go
+++ b/vendor/github.com/opencontainers/runc/libcontainer/devices/devices.go
@@ -31,7 +31,7 @@ func DeviceFromPath(path, permissions st
 
 	var (
 		devNumber = stat.Rdev
-		major     = unix.Major(devNumber)
+		major     = unix.Major(uint64(devNumber))
 	)
 	if major == 0 {
 		return nil, ErrNotADevice
@@ -51,7 +51,7 @@ func DeviceFromPath(path, permissions st
 		Type:        devType,
 		Path:        path,
 		Major:       int64(major),
-		Minor:       int64(unix.Minor(devNumber)),
+		Minor:       int64(unix.Minor(uint64(devNumber))),
 		Permissions: permissions,
 		FileMode:    os.FileMode(mode),
 		Uid:         stat.Uid,
