--- a/pkg/signal/signal_linux.go
+++ b/pkg/signal/signal_linux.go
@@ -30,7 +30,6 @@ var SignalMap = map[string]syscall.Signa
 	"PWR":      syscall.SIGPWR,
 	"QUIT":     syscall.SIGQUIT,
 	"SEGV":     syscall.SIGSEGV,
-	"STKFLT":   syscall.SIGSTKFLT,
 	"STOP":     syscall.SIGSTOP,
 	"SYS":      syscall.SIGSYS,
 	"TERM":     syscall.SIGTERM,
@@ -38,7 +37,6 @@ var SignalMap = map[string]syscall.Signa
 	"TSTP":     syscall.SIGTSTP,
 	"TTIN":     syscall.SIGTTIN,
 	"TTOU":     syscall.SIGTTOU,
-	"UNUSED":   syscall.SIGUNUSED,
 	"URG":      syscall.SIGURG,
 	"USR1":     syscall.SIGUSR1,
 	"USR2":     syscall.SIGUSR2,
