--- a/Makefile
+++ b/Makefile
@@ -107,3 +107,17 @@ clean:
 	cd $(FRONTEND_DIR) && $(MAKE) clean
 
 .PHONY: all style format build test vet assets tarball docker promu proto staticcheck
+
+define install_binaries_gopath
+	mkdir -p $(GOPATH)/$(GO_PKG_BIN_PATH)
+	(for binary in alertmanager amtool ; do \
+		( @echo "$$binary"; cp -f ./$$binary \
+			$(GOPATH)/$(GO_PKG_BIN_PATH)/ ); \
+	done; )
+endef
+
+.PHONY: install-gopath
+install-gopath:
+	@echo ">> installing binaries in GOPATH"
+	$(call install_binaries_gopath,$(GOOS),$(GOARCH))
+
