--- a/Makefile
+++ b/Makefile
@@ -103,4 +103,15 @@ $(FIRST_GOPATH)/bin/staticcheck:
 $(FIRST_GOPATH)/bin/govendor:
 	@GOOS= GOARCH= $(GO) get -u github.com/kardianos/govendor
 
-.PHONY: all style check_license format build test vet assets tarball docker promu staticcheck $(FIRST_GOPATH)/bin/staticcheck govendor $(FIRST_GOPATH)/bin/govendor
+define install_binaries_gopath
+	mkdir -p $(GOPATH)/$(GO_PKG_BIN_PATH)
+	(for binary in prometheus promtool ; do \
+		( @echo "$$binary"; cp -f ./$$binary \
+			$(GOPATH)/$(GO_PKG_BIN_PATH)/ ); \
+	done; )
+endef
+
+install-gopath:
+	@echo ">> installing binaries in GOPATH"
+	$(call install_binaries_gopath,$(GOOS),$(GOARCH))
+
+.PHONY: all style check_license format build test vet assets tarball docker promu staticcheck $(FIRST_GOPATH)/bin/staticcheck govendor $(FIRST_GOPATH)/bin/govendor install-gopath
