--- a/vendor/gopkg.in/lxc/go-lxc.v2/lxc-binding.h
+++ b/vendor/gopkg.in/lxc/go-lxc.v2/lxc-binding.h
@@ -73,12 +73,12 @@ extern bool go_lxc_restore(struct lxc_co
  * compilation handle things, but the call will still fail
  */
 #if LXC_VERSION_MAJOR != 2
-struct migrate_opts {
+/*struct migrate_opts {
 	char *directory;
 	bool verbose;
 	bool stop;
 	char *predump_dir;
-};
+};*/
 #endif
 
 /* This is a struct that we can add "extra" (i.e. options added after 2.0.0)
