--- a/epan/dissectors/Makefile.am
+++ b/epan/dissectors/Makefile.am
@@ -1930,6 +1930,12 @@ make_dissectors_CFLAGS = $(WS_CPPFLAGS)
 make_dissectors_LDADD = $(GLIB_LIBS) \
 			${top_builddir}/wsutil/libwsutil.la
 
+make_dissectors_LDFLAGS = -Wl,-rpath,$(STAGING_DIR)/lib \
+			  -Wl,-rpath,$(STAGING_DIR)/usr/lib \
+			  -L$(STAGING_DIR)/lib \
+			  -L$(STAGING_DIR)/usr/lib \
+			  -Wl,--dynamic-linker=$(STAGING_DIR)/lib/loader
+
 # include asn1 in the source tarball. But nothing to build...
 DIST_SUBDIRS = asn1 dcerpc
 
